.github/workflows/zero-downtime-tests.yml

```
@@ -313,50 +306,98 @@ jobs:
       - name: Prepare test run
         id: prepare_test_run
         working-directory: test/zero-downtime
-        run: go get ./...
+        run: |
+         mkdir ../../temp_res
+         go install github.com/jstemmer/go-junit-report/v2@v2.0.0
+         go get ./...
 
       #######################################################################
       # TESTS
       #######################################################################
 
       - name: Zero Downtime Tests
-        timeout-minutes: 90
+        timeout-minutes: 30
         working-directory: test/zero-downtime
         id: test_zero_downtime
         env:
           GOMAXPROCS: 10
           KEPTN_ENDPOINT: ${{ steps.determine_keptn_endpoint.outputs.KEPTN_ENDPOINT }}
         run: |
-          set -euo pipefail 
           
-          go test -json -v -run Test_ZeroDowntime -v -timeout 90m 2>&1 | tee /tmp/gotest.log & GOTEST=$!
-
+          go test -v -json -run Test_ZeroDowntime -timeout 25m > ../../temp_res/gotest.log & GOTEST=$!
+          
           cd ../../bridge &&
           if [[ "${{ github.event.inputs.testUI }}" == 'true' ]]; then
             while [[ $(jobs -pr) ]]; do
-              echo "New Run: "  >> /tmp/uitest.log 
-              npm run test:ui >> /tmp/uitest.log 2>&1 
+              echo "New Run: " >> /tmp/uitest.log
+              npm run test:ui >> /tmp/uitest.log 2>&1
             done
           else
             echo "UI tests will not run"
           fi
         
           wait $GOTEST
           if [[ $? -ne 0 ]]; then
-            echo "::error::Zero Downtime go test failed - exiting"
+            echo " Zero Downtime go test failed - exiting "
+            echo "::error::Zero Downtime go test failed - exiting"          
             exit 1
           fi
+          echo "Zero Downtime go test finished, storing result"
+
+      - name: Generate XML
+        if: always()
+        run: go-junit-report -in temp_res/gotest.log -parser gojson -out temp_res/gotest.xml
+
+      - name: Generate test Summary
+        uses: test-summary/action@v1
+        with:
+          paths: "temp_res/*.xml"
+          output: temp_res/test-summary.md
+        if: always()
```

We always have the `if` at the top :)
```suggestion
      - name: Generate Test Summary
        if: always()
        uses: test-summary/action@v1
        with:
          paths: "temp_res/*.xml"
          output: temp_res/test-summary.md
```